### DISCUSSION TITLE
How to catch exceptions with the transaction api

### DISCUSSION THREAD

**Discussion by vladwulf** [prisma/prisma#7076] [Q&A]
Hey guys,

I've been going through the exceptions catching section here https://www.prisma.io/docs/concepts/components/prisma-client/handling-exceptions-and-errors

This does not work when I use the transaction API.
So if I try to create a user with a duplicate email address and that I have more than 2 transactions in the transaction array of the transaction API I get the following error.

```
Error: Error occurred during query execution:
ConnectorError(ConnectorError { user_facing_error: Some(KnownError { message: "Unique constraint failed on the fields: (`email`)", meta: Object({"target": Array([String("email")])}), error_code: "P2002" }), kind: UniqueConstraintViolation { constraint: Fields(["email"]) } })
    at /Users/rocinante/code/side/talo/talo-api/node_modules/@prisma/client/runtime/index.js:27757:19
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
```

Any tips to catch exceptions based on Prisma codes with a transaction?

**Comment by janpio** [comment]
Can you share the code snippet you are using to cause this? Would be helpful if we could just reproduce the behavior you are seeing.
